src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -300,11 +306,38 @@ class AppsResource @Inject() (
   private def maybePostEvent(req: HttpServletRequest, app: AppDefinition) =
     eventBus.publish(ApiPostEvent(req.getRemoteAddr, req.getRequestURI, app))
 
-  private def search(cmd: String, id: String): Iterable[AppDefinition] = {
+  type LabelMatcher = (Map[String, String]) => Boolean
+
+  private def labelMatcher(labels: String): LabelMatcher = {
```

Comment:
It would be nice to document your intention. When should the LabelMatcher and when not?